src/context/volume-context.js

```diff
@@ -0,0 +1,82 @@
+import {createContext, useContext, useReducer} from "react";
+
+const VolumeContext = createContext();
+
+// Custom component to provide the Volume context
+function VolumeProvider(props) {
+  const [state, dispatch] = useReducer(volumeReducer, {
+    toggles: {
+      measurement: 0,
+      season: 0, 
+      tide: 0,
+    },
+    path: "./assets/models/summer_low_salt.png",
+  })
+
+  const value = {state, dispatch}
+  return (
+    <VolumeContext.Provider value={value}>
+      {props.children}
+    </VolumeContext.Provider>
+  );
+}
+
+// Custom hook to get the current VolumeContext 
+function useVolume() {
+  const context = useContext(VolumeContext);
+  if (context === undefined) {
+    throw new Error("useVolume must be used within a VolumeProvider")
+  }
+  return context;
+}
+
+// Custom reducer to update the VolumeContext
+function volumeReducer(state, action) {
+  function getPath() {
+    let path = "";
```

Comment:
once you change the `toggles above`, this can become
```
const path = `${state.blah.season}_${state.blah.tide}_${state.blah.measurement}.png`
```

Reply:
the `state.asset_path` could also be interpolated into the string here

Reply:
By `state.asset_path` do you mean the `./assets/models/` prefix?

Reply:
yeah, though given the unavailability of `useEffect` here, your current approach is good for the moment. I would just make sure that the path is always in sync with the options (maybe call `getPath` early, or otherwise set the initial values to variables and construct the initial state from those variables instead of hardcoded.